pkg/net/net.go

```diff
@@ -24,7 +24,7 @@ func IPAddrs() (ips []net.IP, err error) {
 	}
 
 	for _, a := range addrs {
-		if ipnet, ok := a.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+		if ipnet, ok := a.(*net.IPNet); ok && ipnet.IP.IsGlobalUnicast() {
```

Comment:
this is a major change here, the problem was that once we assign `link-local` address 169.254.x.x to the VM in network config, it gets picked up as node address, and this kills the etcd cluster obviously.

`IsGlobalUnicast` is defined as the following: 

```
	return (len(ip) == IPv4len || len(ip) == IPv6len) &&
		!ip.Equal(IPv4bcast) &&
		!ip.IsUnspecified() &&
		!ip.IsLoopback() &&
		!ip.IsMulticast() &&
		!ip.IsLinkLocalUnicast()
```

it fiters out multicast, link-local, loopback, unspecified (zeroes), but it allows for private and any public IPs. this seems to be a reasonable change, but still might be a major change